packages/bot-skeleton/src/scratch/utils/index.js

```diff
@@ -7,6 +7,53 @@ import { removeLimitedBlocks } from '../../utils/workspace';
 import { saveWorkspaceToRecent } from '../../utils/local-storage';
 import DBotStore from '../dbot-store';
 import { log_types } from '../../constants/messages';
+import { error_message_map } from '../../utils/error-config';
+
+export const matchTranslateAttribute = translateString => {
+    const match = translateString.match(/translate\((-?\d*\.?\d+),(-?\d*\.?\d+)\)/);
+
+    if (match) {
+        const x = parseFloat(match[1]);
+        const y = parseFloat(match[2]);
+        return { x, y };
+    }
+    return null; // Invalid or no match
+};
+
+export const extractTranslateValues = () => {
+    const transform_value = Blockly?.derivWorkspace?.trashcan.svgGroup_.getAttribute('transform');
+    const translate_xy = matchTranslateAttribute(transform_value);
+
+    if (!translate_xy) {
+        globalObserver.emit('Error', 'Invalid String');
+    }
+
+    return {
+        translate_X: translate_xy.x,
+        translate_Y: translate_xy.y,
+    };
+};
+
+export const validateErrorOnBlockDelete = () => {
+    // Get the bounding rectangle of the selected block
+    const mandatory_blocks = ['trade_definition', 'before_purchase', 'during_purchase'];
+    const { translate_X, translate_Y } = extractTranslateValues();
+    const blockRect = Blockly.selected?.getSvgRoot().getBoundingClientRect();
+
+    // Extract coordinates from the bounding rectangles
+    const blockX = blockRect?.left || 0;
+    const blockY = blockRect?.top || 0;
+    if (mandatory_blocks?.includes(Blockly?.selected?.type)) {
+        if (
+            blockY >= translate_Y - 100 &&
+            blockY <= translate_Y + 200 &&
+            blockX >= translate_X - 100 &&
+            blockX <= translate_X + 200
```

Comment:
Instead of setting the range 100 to 200 can we stick to one constant value like 200 here? And also take it from a constant variable to later change from one place easily?

Reply:
done ✅